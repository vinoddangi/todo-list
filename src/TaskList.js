import { List, ListItem, IconButton, ListItemText, Stack } from '@mui/material';
import React from 'react';
import { entitySelector, addToList } from './store/entity/entitySlice';
import { useSelector, useDispatch } from 'react-redux';

import { FavoriteOutlined, DeleteOutline } from '@mui/icons-material';

import { pink } from '@mui/material/colors';

function TaskList() {
  const todos = useSelector(entitySelector('todos'));
  return (
    <List sx={{ width: '100%', height: '600px', overflowY: 'auto', scrollbarWidth: 'thin' }}>
      {todos.map(({ text, favorite }, index) => (
        <ListItem
          key={`${text}-${index}`}
          secondaryAction={
            <Stack direction="row" edge="end" spacing={1}>
              <IconButton edge="end" aria-label="Favorite">
                <FavoriteOutlined sx={{ color: favorite ? pink[500] : 'inherit' }} />
              </IconButton>
              <IconButton edge="end" aria-label="Favorite">
                <DeleteOutline />
              </IconButton>
            </Stack>
          }
          sx={{ width: '100%', bgcolor: 'background.paper', color: 'black', marginTop: '4px' }}
        >
          <ListItemText id={text} primary={text} />
        </ListItem>
      ))}
    </List>
  );
}
export { TaskList };
