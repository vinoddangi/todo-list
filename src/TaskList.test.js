import renderer from 'react-test-renderer';
import React from 'react';
import { TaskList } from './TaskList';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
afterAll(() => jest.clearAllMocks());
describe('Given TaskList', () => {
  let tree;

  beforeEach(() => {
    const component = renderer.create(<TaskList />);
    tree = component.toJSON();
  });

  test('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
