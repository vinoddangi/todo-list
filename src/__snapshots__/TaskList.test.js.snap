// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given TaskList renders correctly 1`] = `
<ul
  className="MuiList-root MuiList-padding css-62wvau-MuiList-root"
/>
`;
